storage/paths/index.go

```diff
@@ -604,6 +699,52 @@ func (i *MemIndex) StorageBestAlloc(ctx context.Context, allocate storiface.Sect
 			continue
 		}
 
+		if len(p.info.AllowMiners) > 0 {
+			found := false
+			for _, m := range p.info.AllowMiners {
+				minerIDStr := m
+				maddr, err := address.NewFromString(minerIDStr)
+				if err != nil {
+					return nil, xerrors.Errorf("parsing miner address: %w", err)
+				}
+				mid, err := address.IDFromAddress(maddr)
+				if err != nil {
+					return nil, xerrors.Errorf("converting miner address to ID: %w", err)
+				}
+				if abi.ActorID(mid) == miner {
+					found = true
+					break
+				}
+			}
+			if !found {
+				log.Debugf("not allocating on %s, miner %s not allowed", p.info.ID, miner.String())
+				continue
+			}
+		}
+
+		if len(p.info.DenyMiners) > 0 {
+			found := false
+			for _, m := range p.info.DenyMiners {
```

Comment:
This piece of code is written 6 times. Avoiding duplication keeps the code readable.
could this be a utility function: 

if addressutil.Contains(list, address)

Also, you could write it with lo.SomeBy() which does part of this logic.

Reply:
Refactored to use a function